build.gradle.kts

```diff
@@ -224,3 +224,21 @@ tasks.register("printSdkDebugRuntimeClasspath") {
         File("sdk_classpath").writeText(result.joinToString(File.pathSeparator) { it.absolutePath })
     }
 }
+
+tasks.register("listAllPublishedArtifactIds") {
+    doLast {
+        val artifactIds = rootProject.subprojects.flatMap { subproject ->
+            val publishing = subproject.extensions.findByName("publishing") as? PublishingExtension
```

Comment:
minor: `subproject.extensions.findByType<PublishingExtension>()` can be probably simpler